dwl.c

```diff
@@ -541,19 +566,37 @@ createnotifyx11(struct wl_listener *listener, void *data)
 	struct wlr_xwayland_surface *xwayland_surface = data;
 	Client *c;
 
-	/* Allocate a Client for this surface */
-	c = xwayland_surface->data = calloc(1, sizeof(*c));
-	c->xwayland_surface = xwayland_surface;
-	c->isx11 = 1;
-	c->bw = borderpx;
+	if (xwayland_surface->override_redirect) {
 
-	/* Listen to the various events it can emit */
-	c->map.notify = maprequest;
-	wl_signal_add(&xwayland_surface->events.map, &c->map);
-	c->unmap.notify = unmapnotify;
-	wl_signal_add(&xwayland_surface->events.unmap, &c->unmap);
-	c->destroy.notify = destroynotify;
-	wl_signal_add(&xwayland_surface->events.destroy, &c->destroy);
+		/* Allocate an independent for this surface */
+		c = xwayland_surface->data = calloc(1, sizeof(*c));
+		c->xwayland_surface = xwayland_surface;
+
+		/* Listen to the various events it can emit */
+		c->map.notify = maprequestindependent;
+		wl_signal_add(&xwayland_surface->events.map, &c->map);
+		c->unmap.notify = unmapnotifyindependent;
+		wl_signal_add(&xwayland_surface->events.unmap, &c->unmap);
+		c->destroy.notify = destroynotifyindependent;
```

Comment:
Currently a duplicate of `destroynotify` - can we use that for both?

Reply:
Great catch!

Removed the dupe.